### ISSUE TITLE
Limitation of input size

### ISSUE THREAD

**Issue by orca0928** [cqylunlun/GLASS#27]
Thanks for a great repo.
I read issue https://github.com/cqylunlun/GLASS/issues/6#issuecomment-2264916828 and https://github.com/cqylunlun/GLASS/issues/9#issue-2476650445 to change input size.
Thus, I changed 'transforms.Resize(self.resize)' in mvtec.py to ‘transforms.Resize([self.resize, self.resize])’ and comment out 'transforms.CenterCrop(self.imgsize)' in mvtec.py (3 locations each).
I got an error when the input size was 500 but was able to run it when the input size was 512.
I've noticed that after changing the input size in various ways, is the input size can be used only in multiples of 32?
The image I have now is 500 pixels, so I have no problem using it resized to 480 pixels, but I was curious, so I asked the question.

**Comment by cqylunlun** [comment]
Thank you for your recognition. Your judgment is correct—the input size limitation is due to the 32x downsampling required by `WideResNet-50`, as the input size for this backbone must be a multiple of 32.

**Comment by orca0928** [comment]
Thank you for answering.I wasn't familiar with backbone.

I have read the readme and the issue and have a question about the dataset.
ground truth :This is used to calculate pixel_auroc, pixel_ap, pixel_pro.
                       Even if my custom dataset doesn’t have ground truth data, no significant impact on the learning of the model.
foreground mask : Like ground truth,no significant impact on the learning of the model even without this.
dtd : It supports learning of the model and It is also required when using not only MVTec AD, but also my custom dataset.
Is this correct interpretation?

Sorry to ask such a basic question, following on from the previous one, but please answer!

**Comment by cqylunlun** [comment]
No problem at all! I'm happy to help with your questions.
1. **GT:** Yes, in this case, it’s best to conduct experiments to determine a suitable number of epochs to prevent model overfitting.
2. **Fg Mask:** Mostly correct—our experiments on MVTec AD show that its impact is minimal.
3. **DTD:** Yes, this dataset provides the necessary anomaly textures, and we strongly recommend utilizing it across all datasets.

**Comment by orca0928** [comment]
Your answers have been very helpful.

Is ground truth data to prevent model overfitting?
Because of my recent start in machine learning, I don't understand much importance of ground truth data and difference between image_auroc and pixel_auroc.

Are you saying that "it's best to conduct experiments to determine a suitable number of epochs" means best_record(ckpt_best_epoch.pth file) is not always best model if there is no ground truth data?
Next time I run the program, I will save ckpt.pth file every 10 epochs in addition to the best_record.

**Comment by cqylunlun** [comment]
Ground truth data is not used in training but rather as a validation set to achieve the best performance at a specific epoch, which is a common paradigm in the field of anomaly detection. Without ground truth data, the model may overfit, leading to suboptimal performance at certain epochs. Therefore, your strategy of evaluating every 10 epochs is feasible.

**Comment by orca0928** [comment]
I learn so much talking to you.
If I want to get groundtruth data, is there a way to generate it automatically?
Is it common to obtain it by hand?
Sorry for the question that has nothing to do with GLASS.

**Comment by cqylunlun** [comment]
You're welcome. Manual annotation is the most reliable way to obtain ground truth, and you can use tools like `Labelme` to accomplish this. Of course, if you only need a rough mask, you could also binarize the output results from Glass to create an automatically rough-annotated test set.

**Comment by orca0928** [comment]
Thanks a lot.
All my questions have been resolved.
